src/App.tsx

```diff
@@ -1,24 +1,73 @@
 import React from 'react';
+import { Clock } from './components/Clock';
 import './App.scss';
 
-const App: React.FC = () => {
-  setInterval(() => {
-    const date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+interface State {
+  isClockVisible: boolean;
+  clockName: number;
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    isClockVisible: true,
+    clockName: 0,
+  };
+
+  randomClockName = () => {
+    this.setState({
+      clockName: Math.ceil(Math.random() * 100),
+    });
+  };
+
+  showClock = () => {
```

Comment:
I would suggest creating one method instead of these two: showClock and hideClock.
Since you are changing the boolean value, as the opposite, you may use "!isClockVisible" in order to change the value of the variable.